doc/source/reference/extensions.rst

```diff
@@ -62,6 +63,26 @@ objects.
       api.extensions.ExtensionArray.ndim
       api.extensions.ExtensionArray.shape
       api.extensions.ExtensionArray.tolist
+      api.extensions.NDArrayBackedExtensionArray.dtype
```

Comment:
we need the subclasses?

Reply:
Yes, the docs build fails with:

<details>
<pre>
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:19: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.dtype'. Check your autosummary_generate setting.
looking for now-outdated files... none found
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.argmax'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.argmin'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.argsort'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.astype'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.dropna'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.equals'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.factorize'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.fillna'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.insert'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.isin'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.isna'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.searchsorted'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.shift'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.take'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.to_numpy'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.tolist'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.unique'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.value_counts'. Check your autosummary_generate setting.
/home/runner/work/pandas/pandas/pandas/core/arrays/_mixins.py:docstring of pandas.core.arrays._mixins.NDArrayBackedExtensionArray:35: WARNING: autosummary: stub file not found 'pandas.api.extensions.NDArrayBackedExtensionArray.view'. Check your autosummary_generate setting.
</pre>
</details>

without these.

See: https://github.com/pandas-dev/pandas/runs/4925388897?check_suite_focus=true#step:5:101

Reply:
I find it strange to have a public class with no reference docs, but I could remove it, especially since we'll be mentioning it in doc/source/development/extending.rst now

Reply:
Ah, I think you need to add the following to the docstring (as is done for PandasArray):

https://github.com/pandas-dev/pandas/blob/4e30caa2ea51213f51e1429381f0e3fd66957c61/pandas/core/arrays/numpy_.py#L47-L53

Reply:
> I find it strange to have a public class with no reference docs,

To be clear, it's not about adding `NDArrayBackedExtensionArray` itself in the autosummary above, but about adding all its methods/attributes